modules/core/test/image/testImageWarpAffine.cpp

```diff
@@ -0,0 +1,250 @@
+/****************************************************************************
+ *
+ * This file is part of the ViSP software.
+ * Copyright (C) 2005 - 2017 by Inria. All rights reserved.
+ *
+ * This software is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * ("GPL") version 2 as published by the Free Software Foundation.
+ * See the file LICENSE.txt at the root directory of this source
+ * distribution for additional information about the GNU GPL.
+ *
+ * For using ViSP with software that can not be combined with the GNU
+ * GPL, please contact Inria about acquiring a ViSP Professional
+ * Edition License.
+ *
+ * See http://visp.inria.fr for more information.
+ *
+ * This software was developed at:
+ * Inria Rennes - Bretagne Atlantique
+ * Campus Universitaire de Beaulieu
+ * 35042 Rennes Cedex
+ * France
+ *
+ * If you have questions regarding the use of this file, please contact
+ * Inria at visp@inria.fr
+ *
+ * This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+ * WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * Description:
+ * Test for vpImageTools::warpAffine() function.
+ *
+ * Author:
+ * Vikas Thamizharasan
+ *
+ *****************************************************************************/
+/*!
+  \example testImageWarpAffine.cpp
+
+  \brief Test the warpAffine function with different types of transformation matrix
+*/
+
+#include <visp3/gui/vpDisplayD3D.h>
+#include <visp3/gui/vpDisplayGDI.h>
+#include <visp3/gui/vpDisplayGTK.h>
+#include <visp3/gui/vpDisplayOpenCV.h>
+#include <visp3/gui/vpDisplayX.h>
+#include <visp3/io/vpImageIo.h>
+#include <visp3/io/vpParseArgv.h>
+
+#include <visp3/core/vpIoTools.h>
+#include <visp3/core/vpImageTools.h>
+
+namespace {
+
+  /*!
+    Set the program options.
+
+    \param argc : Command line number of parameters.
+    \param argv : Array of command line parameters.
+    \param ipath: Input image path.
+    \param ppath: Personal image path.
+    \param opt_no_display : Do not display if set.
+    \param opt_no_click : Do not need click if set.
+    \param opt_angle : Affine transformation angle of rotation.
+    \param opt_scale_x : Affine transformation scale along x-axis.
+    \param opt_scale_y : Affine transformation scale along y-axis.
+    \param opt_translation_x : Affine transformation translation along x-axis.
+    \param opt_translation_y : Affine transformation translation along y-axis.
+    \return false if the program has to be stopped, true otherwise.
+  */
+  bool getOptions(int argc, const char **argv, std::string &opt_ipath, std::string &opt_ppath,
+                  bool &opt_no_display, bool &opt_no_click, float &opt_angle,
+                  float &opt_scale_x, float &opt_scale_y,
```

Comment:
`opt_no_display` and `opt_no_click` should be declared as `int` nor `bool`. I discover that with `bool` it could produce a segfault os OSX.